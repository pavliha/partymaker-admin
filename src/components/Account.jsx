import React, { Component } from 'react'
import userShape from 'shapes/user'
import { object, string } from 'prop-types'
import { IconButton, withStyles, MenuItem, Menu, Typography } from '@material-ui/core'
import AccountIcon from 'mdi-react/AccountCircleIcon'
import classNames from 'classnames'

const styles = {

  root: {
    paddingTop: 3,
    display: 'flex',
    alignItems: 'center',
  },

  avatar: {
    fontSize: '0.9rem',
  },

  userName: {
    cursor: 'pointer',
    marginLeft: 3,
    marginRight: 15,
    fontWeight: 400,
    fontFamily: 'Google Sans',
  },
}

class Account extends Component {

  state = {
    anchorEl: null,
  }

  open = event => {
    this.setState({ anchorEl: event.currentTarget })
  }

  close = () => {
    this.setState({ anchorEl: null })
  }

  logout = () => {
    window.location.replace('/logout')
  }

  render() {
    const { classes, user, className } = this.props
    const { anchorEl } = this.state

    if (!user) return null

    return (
      <div className={classNames([classes.root, className])}>
        <IconButton
          color="inherit"
          aria-owns={anchorEl ? 'user-menu' : undefined}
          aria-haspopup="true"
          onClick={this.open}
        >
          <AccountIcon className={classes.avatar} user={user} />
        </IconButton>
        <Typography className={classes.userName} onClick={this.open}>
          {user.name}
        </Typography>
        <Menu
          id="simple-menu"
          anchorEl={anchorEl}
          open={Boolean(anchorEl)}
          onClose={this.close}
        >
          <MenuItem to="/logout" onClick={this.logout}>Logout</MenuItem>
        </Menu>
      </div>
    )
  }
}

Account.propTypes = {
  classes: object.isRequired,
  className: string,
  user: userShape.isRequired,
}

export default withStyles(styles)(Account)
