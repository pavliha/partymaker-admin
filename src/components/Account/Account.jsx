import React from 'react'
import userShape from 'shapes/user'
import { func, object, shape, string } from 'prop-types'
import { IconButton, Typography } from '@material-ui/core'
import { withStyles } from '@material-ui/styles'
import classNames from 'classnames'
import { actions, connect, select } from 'src/redux'
import LogoutIcon from 'mdi-react/LogoutIcon'

const styles = {

  root: {
    paddingTop: 3,
    display: 'flex',
    alignItems: 'center',
  },

  avatar: {
    fontSize: '0.9rem',
  },

  userName: {
    marginLeft: 3,
    marginRight: 15,
    fontWeight: 400,
    fontFamily: 'Google Sans',
  },

}

const Account = ({ classes, redux: { user, logout }, className }) => {

  if (!user) return null

  return (
    <div className={classNames([classes.root, className])}>
      <Typography className={classes.userName}>{user.name}</Typography>
      <IconButton onClick={logout} color="inherit"><LogoutIcon /></IconButton>
    </div>
  )
}

Account.propTypes = {
  classes: object.isRequired,
  className: string,
  redux: shape({
    user: userShape,
    logout: func,
  })
}

const redux = state => ({
  user: select.auth.user(state),
  logout: actions.auth.logout,
})

export default withStyles(styles)(connect(redux)(Account))
