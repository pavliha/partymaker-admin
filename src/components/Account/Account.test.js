import React from 'react'
import Account from './Account'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'

describe('<Account />', () => {

  const setup = (Component, state) => {
    const mockStore = configureStore([])
    const store = mockStore(state)

    const tree = mount(
      <Provider store={store}>
        {Component}
      </Provider>
    )

    return tree
  }

  it('renders null when unauthorized', () => {
    const state = {
      auth: { user_id: null },
      users: { entities: {} }
    }
    const tree = setup(<Account />, state)
    expect(toJson(tree)).toMatchSnapshot()
  })

  it('renders username when authorized ', () => {
    const state = {
      auth: { user_id: 1 },
      users: { entities: { 1: { id: 1, name: 'Test User' } } }
    }
    const tree = setup(<Account />, state)
    expect(toJson(tree)).toMatchSnapshot()
  })

})
