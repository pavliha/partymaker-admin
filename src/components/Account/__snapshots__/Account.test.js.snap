// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Account /> renders null when unauthorized 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <WithStyles(undefined)>
    <Connect(Account)
      classes={
        Object {
          "avatar": "Connect(Account)-avatar-2",
          "root": "Connect(Account)-root-1",
          "userName": "Connect(Account)-userName-3",
        }
      }
    >
      <Account
        classes={
          Object {
            "avatar": "Connect(Account)-avatar-2",
            "root": "Connect(Account)-root-1",
            "userName": "Connect(Account)-userName-3",
          }
        }
        redux={
          Object {
            "logout": [Function],
            "user": null,
          }
        }
      />
    </Connect(Account)>
  </WithStyles(undefined)>
</Provider>
`;

exports[`<Account /> renders username when authorized  1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <WithStyles(undefined)>
    <Connect(Account)
      classes={
        Object {
          "avatar": "Connect(Account)-avatar-2",
          "root": "Connect(Account)-root-1",
          "userName": "Connect(Account)-userName-3",
        }
      }
    >
      <Account
        classes={
          Object {
            "avatar": "Connect(Account)-avatar-2",
            "root": "Connect(Account)-root-1",
            "userName": "Connect(Account)-userName-3",
          }
        }
        redux={
          Object {
            "logout": [Function],
            "user": Object {
              "id": 1,
              "name": "Test User",
            },
          }
        }
      >
        <div
          className="Connect(Account)-root-1"
        >
          <WithStyles(ForwardRef(IconButton))
            aria-haspopup="true"
            color="inherit"
            onClick={[Function]}
          >
            <ForwardRef(IconButton)
              aria-haspopup="true"
              classes={
                Object {
                  "colorInherit": "MuiIconButton-colorInherit",
                  "colorPrimary": "MuiIconButton-colorPrimary",
                  "colorSecondary": "MuiIconButton-colorSecondary",
                  "disabled": "Mui-disabled",
                  "edgeEnd": "MuiIconButton-edgeEnd",
                  "edgeStart": "MuiIconButton-edgeStart",
                  "label": "MuiIconButton-label",
                  "root": "MuiIconButton-root",
                  "sizeSmall": "MuiIconButton-sizeSmall",
                }
              }
              color="inherit"
              onClick={[Function]}
            >
              <WithStyles(ForwardRef(ButtonBase))
                aria-haspopup="true"
                centerRipple={true}
                className="MuiIconButton-root MuiIconButton-colorInherit"
                disabled={false}
                focusRipple={true}
                onClick={[Function]}
              >
                <ForwardRef(ButtonBase)
                  aria-haspopup="true"
                  centerRipple={true}
                  className="MuiIconButton-root MuiIconButton-colorInherit"
                  classes={
                    Object {
                      "disabled": "Mui-disabled",
                      "focusVisible": "Mui-focusVisible",
                      "root": "MuiButtonBase-root",
                    }
                  }
                  disabled={false}
                  focusRipple={true}
                  onClick={[Function]}
                >
                  <button
                    aria-haspopup="true"
                    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <Memo(AccountCircleIcon)
                        className="Connect(Account)-avatar-2"
                        user={
                          Object {
                            "id": 1,
                            "name": "Test User",
                          }
                        }
                      >
                        <svg
                          className="mdi-icon Connect(Account)-avatar-2"
                          fill="currentColor"
                          height={24}
                          user={
                            Object {
                              "id": 1,
                              "name": "Test User",
                            }
                          }
                          viewBox="0 0 24 24"
                          width={24}
                        >
                          <path
                            d="M12,19.2C9.5,19.2 7.29,17.92 6,16C6.03,14 10,12.9 12,12.9C14,12.9 17.97,14 18,16C16.71,17.92 14.5,19.2 12,19.2M12,5A3,3 0 0,1 15,8A3,3 0 0,1 12,11A3,3 0 0,1 9,8A3,3 0 0,1 12,5M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12C22,6.47 17.5,2 12,2Z"
                          />
                        </svg>
                      </Memo(AccountCircleIcon)>
                    </span>
                    <NoSsr>
                      <WithStyles(undefined)
                        center={true}
                      >
                        <ForwardRef(TouchRipple)
                          center={true}
                          classes={
                            Object {
                              "child": "MuiTouchRipple-child",
                              "childLeaving": "MuiTouchRipple-childLeaving",
                              "childPulsate": "MuiTouchRipple-childPulsate",
                              "ripple": "MuiTouchRipple-ripple",
                              "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                              "rippleVisible": "MuiTouchRipple-rippleVisible",
                              "root": "MuiTouchRipple-root",
                            }
                          }
                        >
                          <span
                            className="MuiTouchRipple-root"
                          >
                            <TransitionGroup
                              childFactory={[Function]}
                              component={null}
                              exit={true}
                            />
                          </span>
                        </ForwardRef(TouchRipple)>
                      </WithStyles(undefined)>
                    </NoSsr>
                  </button>
                </ForwardRef(ButtonBase)>
              </WithStyles(ForwardRef(ButtonBase))>
            </ForwardRef(IconButton)>
          </WithStyles(ForwardRef(IconButton))>
          <WithStyles(ForwardRef(Typography))
            className="Connect(Account)-userName-3"
            onClick={[Function]}
          >
            <ForwardRef(Typography)
              className="Connect(Account)-userName-3"
              classes={
                Object {
                  "alignCenter": "MuiTypography-alignCenter",
                  "alignJustify": "MuiTypography-alignJustify",
                  "alignLeft": "MuiTypography-alignLeft",
                  "alignRight": "MuiTypography-alignRight",
                  "body1": "MuiTypography-body1",
                  "body2": "MuiTypography-body2",
                  "button": "MuiTypography-button",
                  "caption": "MuiTypography-caption",
                  "colorError": "MuiTypography-colorError",
                  "colorInherit": "MuiTypography-colorInherit",
                  "colorPrimary": "MuiTypography-colorPrimary",
                  "colorSecondary": "MuiTypography-colorSecondary",
                  "colorTextPrimary": "MuiTypography-colorTextPrimary",
                  "colorTextSecondary": "MuiTypography-colorTextSecondary",
                  "displayBlock": "MuiTypography-displayBlock",
                  "displayInline": "MuiTypography-displayInline",
                  "gutterBottom": "MuiTypography-gutterBottom",
                  "h1": "MuiTypography-h1",
                  "h2": "MuiTypography-h2",
                  "h3": "MuiTypography-h3",
                  "h4": "MuiTypography-h4",
                  "h5": "MuiTypography-h5",
                  "h6": "MuiTypography-h6",
                  "noWrap": "MuiTypography-noWrap",
                  "overline": "MuiTypography-overline",
                  "paragraph": "MuiTypography-paragraph",
                  "root": "MuiTypography-root",
                  "srOnly": "MuiTypography-srOnly",
                  "subtitle1": "MuiTypography-subtitle1",
                  "subtitle2": "MuiTypography-subtitle2",
                }
              }
              onClick={[Function]}
            >
              <p
                className="MuiTypography-root Connect(Account)-userName-3 MuiTypography-body1"
                onClick={[Function]}
              >
                Test User
              </p>
            </ForwardRef(Typography)>
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Menu))
            anchorEl={null}
            id="simple-menu"
            onClose={[Function]}
            open={false}
          >
            <ForwardRef(Menu)
              anchorEl={null}
              classes={
                Object {
                  "list": "MuiMenu-list",
                  "paper": "MuiMenu-paper",
                }
              }
              id="simple-menu"
              onClose={[Function]}
              open={false}
            >
              <WithStyles(ForwardRef(Popover))
                PaperProps={
                  Object {
                    "classes": Object {
                      "root": "MuiMenu-paper",
                    },
                  }
                }
                anchorEl={null}
                anchorOrigin={
                  Object {
                    "horizontal": "left",
                    "vertical": "top",
                  }
                }
                getContentAnchorEl={[Function]}
                id="simple-menu"
                onClose={[Function]}
                onEntering={[Function]}
                open={false}
                transformOrigin={
                  Object {
                    "horizontal": "left",
                    "vertical": "top",
                  }
                }
                transitionDuration="auto"
              >
                <ForwardRef(Popover)
                  PaperProps={
                    Object {
                      "classes": Object {
                        "root": "MuiMenu-paper",
                      },
                    }
                  }
                  anchorEl={null}
                  anchorOrigin={
                    Object {
                      "horizontal": "left",
                      "vertical": "top",
                    }
                  }
                  classes={
                    Object {
                      "paper": "MuiPopover-paper",
                      "root": "MuiPopover-root",
                    }
                  }
                  getContentAnchorEl={[Function]}
                  id="simple-menu"
                  onClose={[Function]}
                  onEntering={[Function]}
                  open={false}
                  transformOrigin={
                    Object {
                      "horizontal": "left",
                      "vertical": "top",
                    }
                  }
                  transitionDuration="auto"
                >
                  <ForwardRef(Modal)
                    BackdropProps={
                      Object {
                        "invisible": true,
                      }
                    }
                    className="MuiPopover-root"
                    id="simple-menu"
                    onClose={[Function]}
                    open={false}
                  />
                </ForwardRef(Popover)>
              </WithStyles(ForwardRef(Popover))>
            </ForwardRef(Menu)>
          </WithStyles(ForwardRef(Menu))>
        </div>
      </Account>
    </Connect(Account)>
  </WithStyles(undefined)>
</Provider>
`;
